src/include/uuid.h

```diff
@@ -26,8 +27,8 @@ struct uuid_d {
   }
 
   void generate_random() {
-    boost::random::random_device rng("/dev/urandom");
-    boost::uuids::basic_random_generator<boost::random::random_device> gen(&rng);
+    std::random_device rng;
+    boost::uuids::basic_random_generator gen(rng);
```

Comment:
mybe basic_random_generator need a template argument. it should be `boost::uuids::basic_random_generator<std::random_device>`

Reply:
in C++17, it can be deduced. BTW, the change has been merged. you might want to file a ticket a PR if you believe it is a bug.

Reply:
@tchaikov  thanks a lot. the newer the better. we will just fix it at our own branch.